### ISSUE TITLE
provider/aws: Invalid protocol aws_network_acl_rule: Invalid Protocol 47

### ISSUE THREAD

**Issue by certnew** [terraform-providers/terraform-provider-aws#1651]
Hi there,

### Terraform Version
Terraform v0.10.3

### Affected Resource(s)
aws_network_acl_rule

### Terraform Configuration Files
```hcl
resource "aws_network_acl_rule" "260" {
  network_acl_id = "${aws_network_acl.NACL-PUBLIC.id}"
  rule_number    = 260
  egress         = false
  protocol       = 47
  rule_action    = "allow"
  cidr_block     = "0.0.0.0/0"
}
```

### Debug Output
Error applying plan:
1 error(s) occurred:
* aws_network_acl_rule.260: 1 error(s) occurred:
* aws_network_acl_rule.260: Invalid Protocol 47 for rule 260

### Expected Behavior
That the terraform plan completes without giving any errors.

### Actual Behavior
Rule actually created and there is in state file

### Steps to Reproduce
1. `terraform apply`

### References
Similar to https://github.com/hashicorp/terraform/issues/12103

**Comment by jrwilk01** [comment]
Similar issues have been fixed by adding to this method:

https://github.com/terraform-providers/terraform-provider-aws/blob/b8d4e1570fc43a2acee6b6e47f63c9db6b067fa2/aws/network_acl_entry.go#L98

Seems like hard coding magic values in the source will lead to more problems.

**Comment by bflad** [comment]
The full list of protocol names/numbers has been added now via #4956 and will release with version 1.25.0 of the AWS provider, likely middle of this week. While it may not be ideal when new protocol names are assigned, it was a simple change to add support for the above case.

**Comment by bflad** [comment]
This has been released in [version 1.25.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1250-june-27-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
